cmd/nerdctl/run_network_linux_test.go

```diff
@@ -529,3 +529,44 @@ func TestSharedNetworkStack(t *testing.T) {
 	base.Cmd("exec", containerNameJoin, "wget", "-qO-", "http://127.0.0.1:80").
 		AssertOutContains(testutil.NginxAlpineIndexHTMLSnippet)
 }
+
+func TestRunContainerWithMACAddress(t *testing.T) {
+	base := testutil.NewBase(t)
+	tID := testutil.Identifier(t)
+	networkBridge := "testNetworkBridge" + tID
+	networkMACvlan := "testNetworkMACvlan" + tID
+	networkIPvlan := "testNetworkIPvlan" + tID
+	base.Cmd("network", "create", networkBridge, "--driver", "bridge").AssertOK()
+	base.Cmd("network", "create", networkMACvlan, "--driver", "macvlan").AssertOK()
+	base.Cmd("network", "create", networkIPvlan, "--driver", "ipvlan").AssertOK()
+	t.Cleanup(func() {
+		base.Cmd("network", "rm", networkBridge).AssertOK()
+		base.Cmd("network", "rm", networkMACvlan).AssertOK()
+		base.Cmd("network", "rm", networkIPvlan).AssertOK()
+	})
+	tests := []struct {
+		Network string
+		Err     string
+	}{
+		{"host", "conflicting options"},
+		{"none", "can't open '/sys/class/net/eth0/address'"},
+		{"container:whatever" + tID, "conflicting options"},
+		{"bridge", ""},
+		{networkBridge, ""},
+		{networkMACvlan, ""},
+		{networkIPvlan, "not support"},
+	}
+	for _, test := range tests {
+		macAddress, err := nettestutil.GenerateMACAddress()
+		if err != nil {
+			t.Errorf("failed to generate MAC address: %s", err)
+		}
+		cmd := base.Cmd("run", "--rm", "--network", test.Network, "--mac-address", macAddress, testutil.CommonImage, "cat", "/sys/class/net/eth0/address")
```

Comment:
I think a nice test will be to have without network option specified.

Reply:
Practically, this PR is for supporting setting MAC addresses in some network types, so specifying the kinds of networks is necessary.